blog/2020-09-14-debezium-auto-create-topics.adoc

```diff
@@ -0,0 +1,235 @@
+= Debezium Auto Create Topics
+rkerner
+:awestruct-tags: [ kafka, topics, production, news ]
+:awestruct-layout: blog-post
+
+When you are working with Kafka Connect then you might have realized that once you start Kafka Connect
+there are already some internal Kafka Connect topics created for you:
+
+[source,options="nowrap",shell]
+----
+$ kafka-topics.sh --bootstrap-server $HOSTNAME:9092 --list
+
+__consumer_offsets
+connect_configs
+connect_offsets
+connect_statuses
+----
+
+This is done automatically for you by Kafka Connect with a sane, customized default topic configuration
+that fits the needs of these internal topics.
+
+When you start a Debezium connector the topics for the captured events are created by the Kafka
+broker based on a default, maybe customized, configuration in the broker if
+`auto.create.topics.enable = true` is enabled in the broker config:
+
+[source,options="nowrap",shell]
+----
+auto.create.topics.enable = false
+default.replication.factor = 1
+num.partitions = 1
+compression.type = producer
+log.cleanup.policy = delete
+log.retention.ms = 604800000  ## 7 days
+----
+
+But often, when you use Debezium and Kafka in a production environment you might often choose to disable
+Kafka's topic auto creation capability with `auto.create.topics.enable = false`, or you want the
+connector topics to be configured differently from the default. In this case you have to create
+topics for Debezium's captured data sources upfront.{empty} +
+But there's good news! Since Kafka Connect version 2.6.0 and above this can be now automated since
+https://cwiki.apache.org/confluence/display/KAFKA/KIP-158%3A+Kafka+Connect+should+allow+source+connectors+to+set+topic-specific+settings+for+new+topics[KIP-158]
+is implemented to enable customizable topic creation with Kafka Connect.
+
+== Kafka Connect
+
+Kafka Connect since Kafka 2.6.0 comes with topic creation enabled:
+
+[source,options="nowrap",shell]
+----
+topic.creation.enable = true
+----
+
+If you don't want to allow automatic topic creation by connectors you can set this value to `false`
+in the Kafka Connect config (_connect-distributed.properties_ file or via environment variable
+_CONNECT_TOPIC_CREATION_ENABLE_).
```

Comment:
"when using Debezium's container image for Kafka Connect [link to DockerHub]"

Reply:
will do

Reply:
Done